internal/objects/service/service.go

```diff
@@ -49,10 +50,23 @@ const (
 	// TODO [danehans]: Make proxy protocol configurable or automatically enabled. See
 	// https://github.com/projectcontour/contour-operator/issues/49 for details.
 	awsLbBackendProtoAnnotation = "service.beta.kubernetes.io/aws-load-balancer-backend-protocol"
+	// awsLbEipAnnotation is a Service annotation that provides capability to assign Load
+	// Balancer IP based on AWS Elastic IP resource in external LoadBalancer scope
+	awsLbEipAnnotation = "service.beta.kubernetes.io/aws-load-balancer-eip-allocations"
```

Comment:
You are correct, this only works with AWS NLBs.
Can you maybe point me to example that shows only classic LB is supported? I only found that any type of LoadBalancer parameters were so far controlled via annotations (e.g. `service.beta.kubernetes.io/aws-load-balancer-backend-protocol`). Is the reason somewhere here? Or is there lack of support for NLB from Contour itself?
Do you mean that we have to adjust annotations once deploying Envoy service, like described [here](https://projectcontour.io/guides/deploy-aws-nlb/)?

Reply:
Contour can work with NLB but the operator does not currently support creating the Envoy service with the "service.beta.kubernetes.io/aws-load-balancer-type: nlb" annotation. When a service is created without any annotations for the aws cloud provider, a class ELB is created (current implementation). See https://github.com/projectcontour/contour-operator/issues/187 for additional background. As I previously mentioned, NLB supported will be needed before merging this PR due to this dependency.

Reply:
Thanks for explanation. Should I open separate PR for that or can I include changes here?
Basic idea would be to add `loadBalancerType` field to `Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.aws` and in case it is NLB set up annotations correctly.